generator/pkg/schemagen/generate.go

```diff
@@ -218,6 +218,13 @@ func (g *schemaGenerator) javaType(t reflect.Type) string {
 		return g.resolveJavaClassUsingMappingSchema(t)
 	}
 
+	// the interface k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/JSON is sometimes used to represent any JSON value
+	// example: Tekton Triggers -> Trigger.spec.interceptors.params.value
+	// -> fallback to Object
+	if t.PkgPath() == "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1" && t.Name() == "JSON" {
```

Comment:
We're mapping JSON to Jackson's `com.fasterxml.jackson.databind.JsonNode` in kubernetes-model-apiextensions:
https://github.com/fabric8io/kubernetes-client/blob/4dae1319b70ab2d3429508fc31d60fa96449a7bf/kubernetes-model-generator/kubernetes-model-apiextensions/cmd/generate/generate.go#L100-L102

Do you think maybe we should map to `com.fasterxml.jackson.databind.JsonNode` here also?

Reply:
Thanks for the hint, I was not aware of that. Using manualTypeMapping instead of adjusting the generator is much better, I´ve applied this!

I´m not sure about using JsonNode though. Using java.lang.Object enables

```java
new InterceptorParamsBuilder()
 .withName("filter")
 .withValue("header.match('X-GitHub-Event', 'pull_request')")
 .build();

new InterceptorParamsBuilder()
  .withName("secretRef")
  .withValue(Map.of("secretName", "github-secret", "secretKey", "secretToken"))
  .build();
```

instead of having to use
```java
new InterceptorParamsBuilder()
  .withName("filter")
  .withValue(TextNode.valueOf("header.match('X-GitHub-Event', 'pull_request')"))
  .build()
```
... and I would have to look up how to do this for maps or objects.

What do you think about this?

Reply:
hmm, I think `java.lang.Object` seems better to be used here. I think it's fine. Thanks for the explanation!